src/Network/Pusher/Data.hs

```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE RecordWildCards #-}
```

Comment:
While it may seem like there's more typing/boilerplate for the second approach, I prefer it. I think it most accurately models the allowable data. I'm also not sure how the first approach would type check since a Set/List must have elements of the same type.